### PR TITLE
feat: Create search by nickname.

### PR DESCRIPTION
**Pull request by Egorkostuck**
feat: Create search by nickname.

### CODE REVIEW

File: src/components/layout/defaultLayout/DefaultLayout.tsx [hunk 1]
```diff
@@ -16,27 +17,18 @@ import { getIsAuth } from 'store';
 
 export const DefaultLayout: FC = () => {
   const isAuth = useAppSelector(getIsAuth);
-  const [opened, { toggle }] = useDisclosure();
+  const [opened] = useDisclosure();
```
Comment (AdammmVV) [inline src/components/layout/defaultLayout/DefaultLayout.tsx]: what is useDisclosure for?

File: src/components/layout/defaultLayout/DefaultLayout.tsx [hunk 2]
```diff
@@ -16,27 +17,18 @@ import { getIsAuth } from 'store';
...
 
   return (
     <AppShell
       header={{ height: 65 }}
       navbar={{ width: 150, breakpoint: 'sm', collapsed: { mobile: !opened } }}
```
Comment (AdammmVV) [inline src/components/layout/defaultLayout/DefaultLayout.tsx]: use boolean for mobile

File: src/components/shared/searchInput/SearchInput.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    dispatch(searchDataThank(search));
+  };
+
+  const handleSearch = (value: string): void => {
+    combobox.openDropdown();
+    combobox.updateSelectedOptionIndex();
+    setSearch(value);
+  };
+
+  useDebounced({
```
Comment (AdammmVV) [inline src/components/shared/searchInput/SearchInput.tsx]: use mantine hook

File: src/components/shared/searchInput/SearchInput.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            setSearch(value || '');
+          }}
+          placeholder={t('friendSearch')}
+          rightSectionPointerEvents='none'
+        />
+      </Combobox.Target>
+
+      <Combobox.Dropdown>
+        <Combobox.Options>
+          {searchData ? options : <Combobox.Empty>Nothing found</Combobox.Empty>}
```
Comment (AdammmVV) [inline src/components/shared/searchInput/SearchInput.tsx]: add translate

File: src/components/shared/searchInput/SearchInput.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  const [value, setValue] = useState<string | null>(null);
+  const [search, setSearch] = useState('');
+  const { t } = useTranslation();
+
+  const options = searchData?.map((item) => (
+    <Combobox.Option value={item.nickname} key={item.nickname}>
+      <Avatar size={24} src={item.photoURL} radius={24} />
+      <span style={{ marginLeft: 10 }}>{item.nickname}</span>
+    </Combobox.Option>
+  ));
```
Comment (AdammmVV) [inline src/components/shared/searchInput/SearchInput.tsx]: move `map` to markup

File: src/components/shared/searchInput/SearchInput.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+          leftSection={<IconSearch />}
+          value={search}
+          onChange={(event) => handleSearch(event.currentTarget.value)}
+          onClick={() => combobox.openDropdown()}
+          onFocus={() => combobox.openDropdown()}
+          onBlur={() => {
+            combobox.closeDropdown();
+            setSearch(value || '');
+          }}
+          placeholder={t('friendSearch')}
```
Comment (AdammmVV) [inline src/components/shared/searchInput/SearchInput.tsx]: take it out of the markup

File: src/api/api.ts [hunk 7]
```diff
@@ -87,6 +88,34 @@ export const checkFieldValueExists = async (
   return !querySnapshot.empty;
 };
 
+export const searchData = async (
+  collectionName: NamesDBCollection,
+  fieldName: string,
+  value: string,
+): Promise<Array<DocumentData>> => {
+  const collectionRef = collection(db, collectionName);
+  const searchTermLow = value.toLowerCase();
```
Comment (AdammmVV) [inline src/api/api.ts]: in order to search case sensitive, will I need to create another method?

File: src/components/layout/header/Header.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import { SearchInput } from 'components';
+import { useAppSelector } from 'hooks';
+import { getIsAuth } from 'store';
+
+export const Header: FC = () => {
+  const isAuth = useAppSelector(getIsAuth);
+  const [opened, { toggle }] = useDisclosure();
+
+  return (
+    <AppShell.Header p='sm'>
```
Comment (AdammmVV) [inline src/components/layout/header/Header.tsx]: `AppShell` is not related to `Header`

File: src/api/api.ts (hunk 7 above)
Comment (Egorkostuck) [reply src/api/api.ts]: I will fix this defect and make a parameter defining the type of case search.

**Pull request merged**